### PR TITLE
Unify creating a BoxedExentsionFactory

### PR DESCRIPTION
**Pull request by mariaschett**
Unify creating a BoxedExentsionFactory with `new_boxed_extension_factory`

For LookupFactory:
* merge `new` and `create` to `new_boxed_extension_factory`

For PrivateMetricsFactory:
* refactor `create_metics_proxy_factory` and `new` to `new_boxed_extension_factory` by
  * inlining `validate` and boxing in `new_boxed_extension_factory`
  * checking `config` is some before creating the factory
  
For TensorFlowFactory:
* refactor `create_tensorflow_factory` and `new` to `new_boxed_extension_factory` by
  * inlining boxing in `new_boxed_extension_factory`
  * checking `config` is some and async reading the model before creating the factory

### CODE REVIEW

File: oak_functions/examples/fuzzable/module/src/tests.rs [hunk 1]
```diff
@@ -56,7 +56,8 @@ async fn test_server() {
     let logger = Logger::for_test();
 
     let lookup_data = Arc::new(LookupData::new_empty(None, logger.clone()));
-    let lookup_factory = LookupFactory::create(lookup_data, logger.clone()).unwrap();
+    let lookup_factory =
+        LookupFactory::new_boxed_extension_factory(lookup_data, logger.clone()).unwrap();
```
Comment (rbehjati) [inline oak_functions/examples/fuzzable/module/src/tests.rs:60]: pre-existing: Why do we call `unwrap` here, but `expect` for creating `PrivateMetricsProxyFactory`? I think we want to call `expect` here too.

File: oak_functions/examples/key_value_lookup/module/src/tests.rs [hunk 2]
```diff
@@ -76,7 +76,8 @@ async fn test_server() {
         logger.clone(),
     ));
     lookup_data.refresh().await.unwrap();
-    let lookup_factory = LookupFactory::create(lookup_data, logger.clone()).unwrap();
+    let lookup_factory =
+        LookupFactory::new_boxed_extension_factory(lookup_data, logger.clone()).unwrap();
```
Comment (rbehjati) [inline oak_functions/examples/key_value_lookup/module/src/tests.rs:80]: `expect` instead of `unwrap`? And in a few other places too.

**Review by rbehjati — APPROVED** [review]

File: oak_functions/examples/fuzzable/module/src/tests.rs (hunk 1 above)
Comment (mariaschett) [reply oak_functions/examples/fuzzable/module/src/tests.rs]: done.

File: oak_functions/examples/key_value_lookup/module/src/tests.rs (hunk 2 above)
Comment (mariaschett) [reply oak_functions/examples/key_value_lookup/module/src/tests.rs]: done (and in the other places :))

File: oak_functions/examples/weather_lookup/module/src/tests.rs [hunk 3]
```diff
@@ -95,7 +95,8 @@ async fn test_server() {
     ));
     lookup_data.refresh().await.unwrap();
 
-    let lookup_factory = LookupFactory::create(lookup_data, logger.clone()).unwrap();
+    let lookup_factory =
+        LookupFactory::new_boxed_extension_factory(lookup_data, logger.clone()).unwrap();
```
Comment (mariaschett) [inline oak_functions/examples/weather_lookup/module/src/tests.rs]: changed to expect.

File: oak_functions/examples/key_value_lookup/module/src/tests.rs [hunk 4]
```diff
@@ -136,7 +137,8 @@ fn bench_wasm_handler(bencher: &mut Bencher) {
     };
 
     let lookup_data = Arc::new(LookupData::for_test(entries));
-    let lookup_factory = LookupFactory::create(lookup_data, logger.clone()).unwrap();
+    let lookup_factory =
+        LookupFactory::new_boxed_extension_factory(lookup_data, logger.clone()).unwrap();
```
Comment (mariaschett) [inline oak_functions/examples/key_value_lookup/module/src/tests.rs]: changed to expect.

File: oak_functions/examples/weather_lookup/module/src/tests.rs [hunk 5]
```diff
@@ -207,7 +208,8 @@ fn bench_wasm_handler(bencher: &mut Bencher) {
 
     let lookup_data = Arc::new(LookupData::for_test(entries));
     let logger = Logger::for_test();
-    let lookup_factory = LookupFactory::create(lookup_data, logger.clone()).unwrap();
+    let lookup_factory =
+        LookupFactory::new_boxed_extension_factory(lookup_data, logger.clone()).unwrap();
```
Comment (mariaschett) [inline oak_functions/examples/weather_lookup/module/src/tests.rs]: changed to expect

File: oak_functions/loader/fuzz/fuzz_targets/wasm_invoke.rs [hunk 6]
```diff
@@ -116,7 +116,8 @@ fuzz_target!(|instruction_list: Vec<ArbitraryInstruction>| {
 
     let logger = Logger::for_test();
     let lookup_data = Arc::new(LookupData::for_test(entries));
-    let lookup_factory = LookupFactory::create(lookup_data, logger.clone()).unwrap();
+    let lookup_factory =
+        LookupFactory::new_boxed_extension_factory(lookup_data, logger.clone()).unwrap();
```
Comment (mariaschett) [inline oak_functions/loader/fuzz/fuzz_targets/wasm_invoke.rs]: changed to expect.

File: oak_functions/loader/src/tests.rs [hunk 7]
```diff
@@ -147,7 +147,8 @@ where
     lookup_data.refresh().await.unwrap();
     let tee_certificate = vec![];
 
-    let lookup_factory = LookupFactory::create(lookup_data.clone(), logger.clone()).unwrap();
+    let lookup_factory =
+        LookupFactory::new_boxed_extension_factory(lookup_data.clone(), logger.clone()).unwrap();
```
Comment (mariaschett) [inline oak_functions/loader/src/tests.rs]: changed to expect.

File: oak_functions/sdk/oak_functions/tests/integration_test.rs [hunk 8]
```diff
@@ -38,7 +38,8 @@ lazy_static! {
 async fn test_read_write() {
     let logger = Logger::for_test();
     let lookup_data = Arc::new(LookupData::for_test(hashmap! {}));
-    let lookup_factory = LookupFactory::create(lookup_data, logger.clone()).unwrap();
+    let lookup_factory =
+        LookupFactory::new_boxed_extension_factory(lookup_data, logger.clone()).unwrap();
```
Comment (mariaschett) [inline oak_functions/sdk/oak_functions/tests/integration_test.rs]: changed to expect

File: oak_functions/sdk/oak_functions/tests/integration_test.rs [hunk 9]
```diff
@@ -54,7 +55,8 @@ async fn test_read_write() {
 async fn test_double_read() {
     let logger = Logger::for_test();
     let lookup_data = Arc::new(LookupData::for_test(hashmap! {}));
-    let lookup_factory = LookupFactory::create(lookup_data, logger.clone()).unwrap();
+    let lookup_factory =
```
Comment (mariaschett) [inline oak_functions/sdk/oak_functions/tests/integration_test.rs]: changed to expect

File: oak_functions/sdk/oak_functions/tests/integration_test.rs [hunk 10]
```diff
@@ -70,7 +72,8 @@ async fn test_double_read() {
 async fn test_double_write() {
     let logger = Logger::for_test();
     let lookup_data = Arc::new(LookupData::for_test(hashmap! {}));
-    let lookup_factory = LookupFactory::create(lookup_data, logger.clone()).unwrap();
+    let lookup_factory =
```
Comment (mariaschett) [inline oak_functions/sdk/oak_functions/tests/integration_test.rs]: changed to expect

File: oak_functions/sdk/oak_functions/tests/integration_test.rs [hunk 11]
```diff
@@ -86,7 +89,8 @@ async fn test_double_write() {
 async fn test_write_log() {
     let logger = Logger::for_test();
     let lookup_data = Arc::new(LookupData::for_test(hashmap! {}));
-    let lookup_factory = LookupFactory::create(lookup_data, logger.clone()).unwrap();
+    let lookup_factory =
+        LookupFactory::new_boxed_extension_factory(lookup_data, logger.clone()).unwrap();
```
Comment (mariaschett) [inline oak_functions/sdk/oak_functions/tests/integration_test.rs]: changed to expect

File: oak_functions/sdk/oak_functions/tests/integration_test.rs [hunk 12]
```diff
@@ -106,7 +110,8 @@ async fn test_storage_get_item() {
 
     let logger = Logger::for_test();
     let lookup_data = Arc::new(LookupData::for_test(entries));
-    let lookup_factory = LookupFactory::create(lookup_data, logger.clone()).unwrap();
+    let lookup_factory =
+        LookupFactory::new_boxed_extension_factory(lookup_data, logger.clone()).unwrap();
```
Comment (mariaschett) [inline oak_functions/sdk/oak_functions/tests/integration_test.rs]: changed to expect.

**Pull request merged**